x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java

```diff
@@ -73,7 +73,7 @@
 public final class IndexPrivilege extends Privilege {
     private static final Logger logger = LogManager.getLogger(IndexPrivilege.class);
 
-    private static final Automaton ALL_AUTOMATON = patterns("indices:*", "internal:transport/proxy/indices:*");
+    private static final Automaton ALL_AUTOMATON = patterns("indices:*", "internal:transport/proxy/indices:*", "internal:data/read/esql/*");
```

Comment:
This has to be here, otherwise the `internal:data/read/esql/*` actions are neither cluster nor index and can't be authorized -- unless I'm missing something else here...